pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -318,14 +386,20 @@ func (plugin *kubenetNetworkPlugin) setup(namespace string, name string, id kube
 	if err != nil {
 		return fmt.Errorf("unable to understand network config: %v", err)
 	}
-	if res.IP4 == nil {
-		return fmt.Errorf("CNI plugin reported no IPv4 address for container %v.", id)
+	//TODO: v1.16 (khenidak) modify cni types
+	// to provide list of ips instead 2 (v4 and v6).
+	// this will make all further call operate on a list of ips
```

Comment:
Instead of this TODO, I would say "update NET_CONFIG_TEMPLATE to CNI version 0.3.0 or later so that we get multiple IP addresses in the returned Result structure". CNI can provide this, but the internal kubenet config never kept up with that.